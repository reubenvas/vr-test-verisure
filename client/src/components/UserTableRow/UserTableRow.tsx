import React from 'react';
import styles from './styles.scss'; // eslint-disable-line @typescript-eslint/no-unused-vars


type propTypes = {
    id: string;
    label: string;
    url: string;
    referenceUrl: string;
}

const UserTableRow = ({
    id, label, url, referenceUrl,
}: propTypes): React.ReactElement<propTypes> => (
    <tr>
        <td><a href="#" data-id={id} className="deleteUser">Delete</a></td>
        <td>{id}</td>
        <td>{label}</td>
        <td>{url}</td>
        <td>{referenceUrl}</td>
    </tr>
);

export default UserTableRow;
