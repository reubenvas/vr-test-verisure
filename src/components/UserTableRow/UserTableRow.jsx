import React from 'react';

const UserTableRow = (props) => {
    return (
        <tr>
            <td><a href="#" data-id={props.id} className="deleteUser">Delete</a></td>
            <td>{props.id}</td>
            <td>{props.firstName}</td>
            <td>{props.lastName}</td>
            <td>{props.email}</td>
        </tr>
    );
};

export default UserTableRow;
