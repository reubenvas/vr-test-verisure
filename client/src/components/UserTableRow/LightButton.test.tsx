import React from 'react';
import renderer from 'react-test-renderer';
import UserTableRow from './UserTableRow';

describe('DarkButton component', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<UserTableRow id='1' firstName='Olle' lastName='Svensson' email='ollesvensson@gmail.com'/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
