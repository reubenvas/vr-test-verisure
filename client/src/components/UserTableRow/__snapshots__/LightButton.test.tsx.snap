// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkButton component snapshot renders 1`] = `
<tr>
  <td>
    <a
      className="deleteUser"
      data-id="1"
      href="#"
    >
      Delete
    </a>
  </td>
  <td>
    1
  </td>
  <td>
    Olle
  </td>
  <td>
    Svensson
  </td>
  <td>
    ollesvensson@gmail.com
  </td>
</tr>
`;
